Controllers/ApplicationController.cs

```diff
@@ -46,38 +52,39 @@ public RedirectResult RequestBalances()
 
             // generate Redirect Url
             var redirectUrl = string.Format("{0}://{1}/{2}", Request.Url.Scheme, Request.Url.Authority, "fetch-balances");
-            
+
             // set CSRF token in browser cookie
             Response.Cookies.Add(new HttpCookie("csrf_token")
             {
                 Value = csrfToken
             });
-            // Create a token request to be stored
-            var tokenRequest = TokenRequest.AccessTokenRequestBuilder(
-                    ResourceType.Accounts, ResourceType.Balances)
-                .SetToMemberId(pfmMember.MemberId())
-                .SetToAlias(pfmMember.GetFirstAliasBlocking())
-                .SetRefId(refId)
-                .SetRedirectUrl(redirectUrl)
-                .SetCsrfToken(csrfToken)
-                .build();
 
-            var requestId = pfmMember.StoreTokenRequestBlocking(tokenRequest);
-
-            //generate the Token request URL to redirect to
-            var tokenRequestUrl = tokenClient.GenerateTokenRequestUrlBlocking(requestId);
-
-            //send a 302 redirect
-            Response.StatusCode = 302;
-            return new RedirectResult(tokenRequestUrl);
+            return await GetPfmMember().FlatMap(mem => mem.GetFirstAlias()
```

Comment:
the indentation of this paragraph is a little confusing, it could be adjusted like this so all the "map"s are aligned:
``` csharp
            return await GetPfmMember()
                .FlatMap(async mem => await mem.StoreTokenRequest(
                    // Create a token request to be stored
                    TokenRequest.AccessTokenRequestBuilder(ResourceType.Accounts, ResourceType.Balances)
                        .SetToMemberId(mem.MemberId())
                        .SetToAlias(await mem.GetFirstAlias())
                        .SetRefId(refId)
                        .SetRedirectUrl(redirectUrl)
                        .SetCsrfToken(csrfToken)
                        .build()))
                // generate the Token request URL to redirect to
                .FlatMap(requestId => tokenClient.GenerateTokenRequestUrl(requestId))
                .Map(url =>
                {
                    // send a 302 redirect
                    Response.StatusCode = 302;
                    return new RedirectResult(url);
                });
```